File: 2021/Python/src/day_01.py
from pathlib import Path

file = open(Path(__file__).parent.parent / "inputs/day_01.txt").read().splitlines()


def day_01():
    result = sum(1 for curr, prev in zip(file[1:], file) if int(curr) > int(prev))
    print("Day 01, Part 1:", result)


def sliding_window(iterable: list, num: int):
    slices = (iterable[i:] for i in range(num))
    for window in zip(*slices):
        yield window


def day_02():
    nums = [int(i) for i in file]
    result = sum(
        1
        for curr, prev in zip(sliding_window(nums[1:], 3), sliding_window(nums, 3))
        if sum(curr) > sum(prev)
    )
    print("Day 01, Part 2:", result)


if __name__ == "__main__":
    day_01()
    day_02()

File: 2021/Rust/src/day_01.rs
use std::fs;

fn part_1(file: &String) {
    let nums: Vec<u32> = file.lines().filter_map(|x| x.parse::<u32>().ok()).collect();

    let result = nums.iter().zip(nums.iter().skip(1));

    println!("Day 0, Part 1: {:?}", result);
}

fn part_2(file: &String) {
    let result = 0;
    println!("Day 0, Part 2: {}", result);
}

fn main() {
    let file = fs::read_to_string("samples/day_01.txt").expect("error reading file");
    part_1(&file);
}
